Refactor/network

OptimusPi opened the pull request.

LeeHorobin committed:
Network Class

Add a network class

LeeHorobin committed:
Beginning network refactor

Started with moving out the create account network logic in to the network class.

This took forever as it involved also modifying the hasher code as a class so that it can be included in mutiple files without throwing linker errors.

Long long way to go...

LeeHorobin committed:
It Works

I'm honestly surprised it compiles :p
But it runs fine too, moved the create account code to the networking class and verified it works, and returns the correct error messages to the player if accoutn creation fails.

LeeHorobin committed:
Minor Changes

Minor formatting changes whilst trying to get my head around the code structure.

LeeHorobin committed:
Move saveInventory

Move the save inventory function to networking

OptimusPi committed:
Merge branch 'development' into refactor/network

LeeHorobin committed:
Refactor Stat Allocation

Move the code for constructing and sending a packet fro stat point allocation to the network class.

LeeHorobin committed:
Update .gitignore

LeeHorobin committed:
Merge branch 'refactor/network' of https://github.com/OptimusPi/SKO into refactor/network

OptimusPi committed:
Update StickKnightsOnline-Client/main.cpp

OptimusPi committed:
Merge branch 'refactor/network' of https://github.com/OptimusPi/SKO into refactor/network

LeeHorobin committed:
Not working - testing only

Doesn't really work properly, buggy, but good progress made moving the network thread out of main.

LeeHorobin committed:
Merge branch 'refactor/network' of https://github.com/OptimusPi/SKO into refactor/network

OptimusPi committed:
Got the game working pretty well so far!

Added MyID to globals
removed hacky networkLock
changed Login to use BReceive instead of networkLock
moved Login to SKO_Network

LeeHorobin committed:
Moved USE_ITEM

Moved the code for using an item to network class.

LeeHorobin committed:
Move MAKE_CLAN

Move the code which sends a clan creation request to the network class.

LeeHorobin committed:
Moved CAST_SPELL

Getting there.

OptimusPi committed:
made helper functions, moved sendTradeItemOffer() over

OptimusPi committed:
moved allt rading functions over

OptimusPi committed:
moved *everything* to network and it works! needs more testing.

OptimusPi committed:
Update main.cpp

LeeHorobin reviewed (commented):
A couple of strange things happening with shops (and probably trading) in this version that need to be ironed out.

OptimusPi committed:
network fixes

LeeHorobin committed:
Fix silly network issue

LeeHorobin reviewed (approved):
Working as it was before the refactor again.

OptimusPi committed:
implemented voradiac function solution for sending packets very easily from the caller

TODO: move this complicated code into a class
TODO: move all the std::string packet = "0" (old-style) code to use the new class
TODO: lots and lots of testing

OptimusPi committed:
finish up converting all calls to new variadic function send()

OptimusPi committed:
Last changes for now

OptimusPi committed:
Update config.ini
